Playback fix and private session support

dhedberg opened the pull request:
Contains a fix for libQtSpotify cutting off the end of tracks and support for private sessions.

This time against the proper branch, I think.

dhedberg committed:
Avoid throwing away decoded but unplayed buffers at the end of a track.

Change tracks on end_of_track callback instead of trying to calculate the
current position; this avoids forcing a track change too early. Also avoid
throwing away data in the buffers when we automatically change to the next
track since we probably want to listen to the tracks to the end.

dhedberg committed:
Add private session support

lukedirtwalker commented:
Hi, thanks for your work. I added some comments/question, please consider and answer them. Cheers

dhedberg committed:
Use better variable names and fix a couple of comments

dhedberg committed:
Let libspotify handle the private session setting for now

Rather than risking having our setting and libspotify disagreeing and surprising
the user, it's sadly safer to leave it all up to libspotify.

dhedberg commented:
Thanks for your review, I've pushed some fixes. I decided to remove the "hacks" around the privacy setting since I don't want to risk having libspotify and libQtSpotify disagree on whether it's activated or not. I could squash the commits and make a new pull request, but maybe this is ok? (Assuming you're happy with the code)

lukedirtwalker commented:
Thanks for the fixes. I will do a quick test and then merge it. Thanks again :)
